src/main/java/com/smartbear/ready/jenkins/JenkinsSoapUIProTestRunner.java

```diff
@@ -297,27 +298,15 @@ public FormValidation doCheckTestSuite(@QueryParameter String value, @QueryParam
         }
 
         public FormValidation doCheckSlmLicenceApiHost(@QueryParameter String value, @QueryParameter String authMethod) {
-            final AuthMethod slmAuthMethod = AuthMethod.valueOf(authMethod);
-            if (StringUtils.isEmpty(value)) {
-                switch (slmAuthMethod) {
-                    case USER_AND_PASSWORD:
-                        return FormValidation.error("Please, enter valid SLM Licence API Host for User And Password authentication method");
-                    case ACCESS_FOR_EVERYONE:
-                        return FormValidation.error("Please, enter valid SLM Licence API Host for Access for Everyone authentication method");
-                }
-            }
-            return FormValidation.ok();
+            return validateEmptyValue(value, AuthMethod.ACCESS_FOR_EVERYONE, authMethod,
+                    SLM_LICENCE_API_HOST, ACCESS_FOR_EVERYONE);
         }
 
         public FormValidation doCheckSlmLicenceApiPort(@QueryParameter String value, @QueryParameter String authMethod) {
             final AuthMethod slmAuthMethod = AuthMethod.valueOf(authMethod);
-            if (!isValidPort(value)) {
-                switch (slmAuthMethod) {
-                    case USER_AND_PASSWORD:
-                        return FormValidation.error("Please, enter valid SLM Licence API Port for User And Password authentication method");
-                    case ACCESS_FOR_EVERYONE:
-                        return FormValidation.error("Please, enter valid SLM Licence API Port for Access for Everyone authentication method");
-                }
+            if (!isValidPort(value) && slmAuthMethod == AuthMethod.ACCESS_FOR_EVERYONE) {
```

Comment:
Port validation is not needed for Username and password authentication, right?

Reply:
port is not mandatory, but you are right - if provided then should be validated....